commands/azureCommands/acr-build.ts

```diff
@@ -0,0 +1,115 @@
+const status = vscode.window.createOutputChannel('ACR Build status');
+
+// Prompts user to select a subscription, resource group, then registry from drop down. If there are multiple folders in the workspace, the source folder must also be selected.
+// The user is then asked to name & tag the image. A build is queued for the image in the selected registry.
+// Selected source code must contain a path to the desired dockerfile.
+export async function queueBuild(dockerFileUri?: vscode.Uri): Promise<void> {
+    //Acquire information from user
+    const subscription: Subscription = await quickPickSubscription();
+
+    const client: ContainerRegistryManagementClient = await AzureUtilityManager.getInstance().getContainerRegistryManagementClient(subscription);
+    const registry: Registry = await quickPickACRRegistry(true);
+
+    const resourceGroupName: string = getResourceGroupName(registry);
+
+    const osPick = ['Linux', 'Windows'].map(item => <IAzureQuickPickItem<string>>{ label: item, data: item });
+    const osType: string = (await ext.ui.showQuickPick(osPick, { 'canPickMany': false, 'placeHolder': 'Select image base OS' })).data;
+
+    const imageName: string = await quickPickNewImageName();
+
+    //Begin readying build
+    status.show();
+
+    let folder: vscode.WorkspaceFolder;
+    if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length === 1) {
+        folder = vscode.workspace.workspaceFolders[0];
+    } else {
+        folder = await vscode.window.showWorkspaceFolderPick();
+    }
+    const dockerItem: Item = await resolveDockerFileItem(folder, dockerFileUri);
+    const sourceLocation: string = folder.uri.path;
+    const tarFilePath: string = getTempSourceArchivePath();
+
+    const uploadedSourceLocation: string = await uploadSourceCode(client, registry.name, resourceGroupName, sourceLocation, tarFilePath, folder);
+    status.appendLine("Uploaded Source Code to " + tarFilePath);
+
+    const runRequest: DockerBuildRequest = {
+        type: 'DockerBuildRequest',
+        imageNames: [imageName],
+        isPushEnabled: true,
+        sourceLocation: uploadedSourceLocation,
+        platform: { os: osType },
+        dockerFilePath: dockerItem.relativeFilePath
+    };
+    status.appendLine("Set up Run Request");
+
+    const run: Run = await client.registries.scheduleRun(resourceGroupName, registry.name, runRequest);
+    status.appendLine("Schedule Run " + run.runId);
+
+    await streamLogs(registry, run, status, client);
+}
+
+async function uploadSourceCode(client: ContainerRegistryManagementClient, registryName: string, resourceGroupName: string, sourceLocation: string, tarFilePath: string, folder: vscode.WorkspaceFolder): Promise<string> {
+    status.appendLine("   Sending source code to temp file");
+    let source: string = sourceLocation.substring(1);
+    let current: string = process.cwd();
+    process.chdir(source);
```

Comment:
I'm also not sure why it changes the working directory and then changes it back. That's why I didn't touch this code. Should I try this without the dir change? Because I just assumed this is a requirement of the API.

Reply:
That would be good, yeah.

Reply:
Hi, Stephen, I was finally able to figure this out.
The change on the directory is because tar does not allow to specify the directory, and we need to upload the build context from where the file was at. So that's why we change the directory for this step and then change it back.

Reply:
let items = await fse.readdir(source);
    items = items.filter(i => !(i in vcsIgnoreList));
    items = items.map(item => path.resolve(source, item));
    // tslint:disable-next-line:no-unsafe-any
    tar.c({}, items).pipe(fse.createWriteStream(tarFilePath));

Reply:
Actually, that didn't work - added them with full paths.  Use this instead:

status.appendLine(" Sending source code to temp file");
let source: string = rootFolder.uri.fsPath;
let items = await fse.readdir(source);
items = items.filter(i => !(i in vcsIgnoreList));
tar.c({ cwd: source }, items).pipe(fse.createWriteStream(tarFilePath));

Reply:
Note: I'm currently getting this trying to quick build (whether it has the changes above or not):
```
Setting up temp file with 'sourceArchive810210.tar.gz' 
   Sending source code to temp file
   Getting Build Source Upload Url 
   Getting blob info from Upload Url 
   Creating Blob Service 
   Creating Block Blob 
Uploaded Source Code to c:/Users/stephwe/AppData/Local/sourceArchive810210.tar.gz
Set up Run Request
Scheduled Run cc4
2018/11/09 00:24:50 Using acb_vol_977c9414-6e78-464d-bbe6-fe1edc746f36 as the home volume
2018/11/09 00:24:50 Setting up Docker configuration...
2018/11/09 00:24:51 Successfully set up Docker configuration
2018/11/09 00:24:51 Logging in to registry: stephwereg.azurecr.io
2018/11/09 00:24:57 Successfully logged in
2018/11/09 00:24:57 Executing step ID: build. Working directory: '', Network: ''
2018/11/09 00:24:57 Obtaining source code and scanning for dependencies...
2018/11/09 00:24:58 Successfully obtained source code and scanned for dependencies
error checking context: 'syntax error in pattern'.
Error: failed to run step ID: build: exit status 1

Run ID: cc4 failed after 14s
```

Reply:
Did you see my last 2 comments?

Reply:
you deleted the process lines. I think that is why. I add them to the code you posted, and I don't get it. try:
    status.appendLine("   Sending source code to temp file");
    let source: string = rootFolder.uri.fsPath;
    let current: string = process.cwd();
    let items = await fse.readdir(source);
    items = items.filter(i => !(i in vcsIgnoreList));
    // tslint:disable-next-line:no-unsafe-any
    tar.c({ cwd: source }, items).pipe(fse.createWriteStream(tarFilePath));
    process.chdir(current);

Reply:
The whole point was to get rid of changing the current dir, added "cwd" arg.  

Something appears to be wrong with my Dockerfile.

Reply:
Was a problem with the .dockerignore file.

Reply:
Yes. I went to see the tar file and you're correct. Everything looks in place without changing directory.